### ISSUE TITLE
ClassifierDL fails to write output log to HDFS

### ISSUE THREAD

**Issue by cguegi** [JohnSnowLabs/spark-nlp#1014]
Text Classification by using [ClassifierDL](https://github.com/JohnSnowLabs/spark-nlp/blob/master/src/main/scala/com/johnsnowlabs/nlp/annotators/classifier/dl/ClassifierDLApproach.scala) fails with `java.io.FileNotFoundException: Failed to append to non-existent file` when `.setEnableOutputLogs(True)` on a Hadoop cluster.

The pipeline works with  `.setEnableOutputLogs(False)`.

## Description
**Environment**
Hadoop cluster: Cloudera
Hadoop version: 3.1.1
Java: openjdk version "1.8.0_252"
Spark version: 2.4.0
Spark NLP version: 2.5.5
Python: 3.6 and Jupyter Notebook

```
# Spark NLP Pipeline
document_assembler = DocumentAssembler() \
    .setInputCol("description") \
    .setOutputCol("document")

use = UniversalSentenceEncoder.load("/user/xyz/tfhub_use_en_2.4.0_2.4_1587136330099") \
 .setInputCols(["document"])\
 .setOutputCol("sentence_embeddings")

classsifierdl = ClassifierDLApproach() \
  .setInputCols(["sentence_embeddings"]) \
  .setOutputCol("class") \
  .setLabelColumn("category") \
  .setMaxEpochs(5) \
  .setEnableOutputLogs(True)

pipeline = Pipeline().setStages([document_assembler, use, classsifierdl])

pipelineModel = pipeline.fit(trainDataset)
```

## Expected Behavior
ClassifierDL writes output log to `hdfs:///user/xyz/annotator_logs/ClassifierDLApproach_random-string.log`

## Current Behavior
```
Py4JJavaError: An error occurred while calling o184.fit.
: java.io.FileNotFoundException: Failed to append to non-existent file /user/xyz/ClassifierDLApproach_162c97e92216.log for client xx.xxx.xx.xxx
	at org.apache.hadoop.hdfs.server.namenode.FSDirAppendOp.appendFile(FSDirAppendOp.java:105)
	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.appendFile(FSNamesystem.java:2705)
	at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.append(NameNodeRpcServer.java:809)
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.append(ClientNamenodeProtocolServerSideTranslatorPB.java:509)
	at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:528)
	at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:1070)
	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:985)
	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:913)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1876)
	at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2882)

	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.hadoop.ipc.RemoteException.instantiateException(RemoteException.java:121)
	at org.apache.hadoop.ipc.RemoteException.unwrapRemoteException(RemoteException.java:88)
	at org.apache.hadoop.hdfs.DFSClient.callAppend(DFSClient.java:1363)
	at org.apache.hadoop.hdfs.DFSClient.append(DFSClient.java:1420)
	at org.apache.hadoop.hdfs.DFSClient.append(DFSClient.java:1390)
	at org.apache.hadoop.hdfs.DistributedFileSystem$5.doCall(DistributedFileSystem.java:436)
	at org.apache.hadoop.hdfs.DistributedFileSystem$5.doCall(DistributedFileSystem.java:432)
	at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
	at org.apache.hadoop.hdfs.DistributedFileSystem.append(DistributedFileSystem.java:444)
	at org.apache.hadoop.hdfs.DistributedFileSystem.append(DistributedFileSystem.java:413)
	at org.apache.hadoop.fs.FileSystem.append(FileSystem.java:1414)
	at com.johnsnowlabs.nlp.util.io.OutputHelper$.writeAppend(OutputHelper.scala:38)
	at com.johnsnowlabs.ml.tensorflow.Logging$class.outputLog(Logging.scala:22)
	at com.johnsnowlabs.ml.tensorflow.TensorflowClassifier.outputLog(TensorflowClassifier.scala:10)
	at com.johnsnowlabs.ml.tensorflow.TensorflowClassifier.train(TensorflowClassifier.scala:65)
	at com.johnsnowlabs.nlp.annotators.classifier.dl.ClassifierDLApproach.train(ClassifierDLApproach.scala:290)
	at com.johnsnowlabs.nlp.annotators.classifier.dl.ClassifierDLApproach.train(ClassifierDLApproach.scala:39)
	at com.johnsnowlabs.nlp.AnnotatorApproach._fit(AnnotatorApproach.scala:55)
	at com.johnsnowlabs.nlp.AnnotatorApproach.fit(AnnotatorApproach.scala:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
```

## Steps to Reproduce

1.
2.
3.
4.

## Your Environment

* Spark NLP version: 2.5.5
* Apache Spark version: 2.4.0
* Java version (java -version): openjdk version "1.8.0_252"
* Setup and installation (Pypi, Conda, Maven, etc.): Jupiter notebook and pip
* Operating System and version: RHEL 7.8
* Link to your project (if any): similar to [nlp-workshop example](https://github.com/JohnSnowLabs/spark-nlp-workshop/blob/master/jupyter/training/english/classification/ClassifierDL_Train_multi_class_news_category_classifier.ipynb)

**Comment by maziyarpanahi** [comment]
Hi @cguegi 

This issue is with HDFS and how it appends to the file, it was fixed in https://github.com/JohnSnowLabs/spark-nlp/pull/999 and will be included in the next release. For the moment you have to turn off the logging to proceed and check the Driver's logs as it will be printing all the stats there as well until the fix is released.

**Comment by cguegi** [comment]
Hi @maziyarpanahi 
Thank you for the fix!
We actually turned off the logging and it worked as expected.

**Comment by maziyarpanahi** [comment]
You are welcome, I'll leave this open until we actually release it and make sure it works properly.

**Comment by maziyarpanahi** [comment]
This should be fixed in 2.6.0, please feel free to reopen it if it still exists.
